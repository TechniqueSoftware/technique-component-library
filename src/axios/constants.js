import { IS_PROD } from '@global/env'

const API_V3_DOMAIN = window.IS_PROD ? 'https://api.club-os.io' : 'https://test-api.club-os.io'
const API_V3_ACCESS_TOKEN_NAME = 'apiV3AccessToken'
const API_V3_ID_TOKEN_NAME = 'apiV3IdToken'

// const CONTEXT_PATH = IS_PROD ? window.CONTEXT_PATH : 'https://test-api.club-os.io'
const CONTEXT_PATH = IS_PROD ? window.CONTEXT_PATH : window.location.origin

export {
  API_V3_DOMAIN,
  API_V3_ACCESS_TOKEN_NAME,
  API_V3_ID_TOKEN_NAME,
  CONTEXT_PATH
}
