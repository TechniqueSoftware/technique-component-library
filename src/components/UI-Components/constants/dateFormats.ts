// Learn about endians: https://en.wikipedia.org/wiki/Date_format_by_country

export const DATE_FORMAT_ISO = 'YYYY-MM-DD';

export const DATE_FORMAT_MIDDLE_ENDIAN = 'MM/DD/YYYY';

export const DATE_FORMAT_LOCAL_DATETIME = 'YYYY-MM-DDTHH:mm:ss.sss';

export const DATE_FORMAT_FULL = 'dddd, MMMM Do, YYYY';
