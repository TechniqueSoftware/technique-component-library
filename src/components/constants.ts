export const MODULE_NAME = {
  HEADERS: 'Headers',
  DIALOGS: 'Dialogs',
  TABS: 'Tabs',
  BUTTONS: 'Buttons',
  PROGRESS: 'Progress',
  CARDS: 'Cards',
  MULTISELECT: 'MultiSelect',
  INPUTS: 'Inputs',
  ICON: 'Icon',
  TABLES: 'Tables',
  SNACKBARS: 'Snackbars',
  NOTIFICATIONS: 'Notifications',
  SURFACES: 'Surfaces',
  BREADCRUMBS: 'Breadcrumbs',
  ERRORS:'Errors',
};
