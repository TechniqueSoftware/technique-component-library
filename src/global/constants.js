export const COLORS = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  INFO: 'info',
  LIGHT: 'light',
  DARK: 'dark',
  GREY: 'grey'
}

export const SIZES = {
  // XS: 'xs',
  SM: 'sm',
  MD: 'md',
  LG: 'lg'
  // XL: 'xl'
}

export const BUTTON_VARIANT = {
  SOLID: 'solid',
  OUTLINE: 'outline',
  LINK: 'link',
  GHOST: 'ghost'
}

export const INPUT_TYPE = {
  EMAIL: 'email',
  NUMBER: 'number',
  DATE: 'date',
  PASSWORD: 'password'
}

export const DEVICES = {
  DESKTOP: 1024,
  TABLET: 768,
  MOBILE: 0
}
