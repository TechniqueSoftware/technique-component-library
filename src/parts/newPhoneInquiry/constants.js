export const EVENT_ICONS = {
  1: 'MEETING',
  2: 'TRAINER',
  3: 'GROUP_CLASS',
  4: 'SALES_LEAD',
  5: 'ORIENTATION',
  6: 'ASSESSMENT',
  7: 'GROUP_TRAINING',
  8: 'GROUP_CLASS'
}
