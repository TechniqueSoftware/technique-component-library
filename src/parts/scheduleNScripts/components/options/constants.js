export const SCRIPT_CONTACT_TYPES = {
  EMAIL: 2,
  SMS: 3,
  NONE: 4
}
