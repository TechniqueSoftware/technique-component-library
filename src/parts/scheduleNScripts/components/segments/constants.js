export const FOLLOWUP_STATUS = {
  APPT_BOOKED: 2,
  MEMBER: 6,
  WEB_LEAD: 11
}
