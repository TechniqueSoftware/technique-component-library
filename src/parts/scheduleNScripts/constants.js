export const TP_ACTION_TYPES = {
  CALL: 1,
  EMAIL: 2,
  TEXT: 3
}

export const SCRIPT_TABNAME = [{
  id: '1',
  name: 'Call',
  className: 'BsTelephone'
}, {
  id: '2',
  name: 'Email',
  className: 'BsEnvelope'
}, {
  id: '3',
  name: 'Text',
  className: 'BsChatDots'
}]

export const SEGMENT_TABNAME = [
  {
    id: '0',
    name: 'Manage Segments'
  },
  {
    id: '1',
    name: 'Copy Segments'
  }
]

export const SNACKBAR_MESSAGES = {
  NEW_SUCCESS: 'Touchpoint has been created successfully!',
  EDIT_SUCCESS: 'Touchpoint has been updated!',
  DELETE_SUCCESS: 'Touchpoint has been deleted successfully!',
  SCRIPT_SUCCESS: 'Script has been saved successfully!',
  SEGMENT_NEW: 'Segment has been created successfully!',
  SEGMENT_EDIT: 'Changes for this segment have been saved successfully!',
  SEGMENT_DELETE: 'Segment has been deleted successfully!'
}

export const STATUSES_WITH_SEGMENTS = [
  { type: 1, id: 11, attributeType: 'Marketing Source', attributeId: 'marketing' }, // Prospecting -> Web Lead
  { type: 1, id: 2, attributeType: 'Marketing Source', attributeId: 'marketing' }, // Prospecting -> Appt Booked
  { type: 1, id: 6, attributeType: 'Membership Type', attributeId: 'membership' } // Member -> Member
]

export const PERMISSION_KEYS = {
  SCHEDULES: {
    1: 'canEditMemberSalesSchedules',
    2: 'canEditPtSchedules',
    3: 'canEditStudioSchedules'
  },
  SCRIPTS: {
    1: 'canEditMemberSalesScripts',
    2: 'canEditPtScripts',
    3: 'canEditStudioScripts'
  }
}
