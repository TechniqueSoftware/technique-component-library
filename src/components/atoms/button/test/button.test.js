import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { COLORS, SIZES, BUTTON_VARIANT } from '@global/constants'

import Button from '..'

const onClick = jest.fn()

describe('<Button />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Button label='Button' onClick={onClick} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('has correct text', () => {
    render(<Button label='Button' onClick={onClick} />)
    expect(screen.getByRole('button')).toBeInTheDocument()
  })

  describe('action', () => {
    it('is disabled', () => {
      render(<Button label='Button' onClick={onClick} disabled />)
      userEvent.click(screen.getByRole('button'))
      expect(onClick).not.toHaveBeenCalled()
    })

    it('calls function on click', () => {
      render(<Button label='Button' onClick={onClick} />)
      userEvent.click(screen.getByRole('button'))
      expect(onClick).toHaveBeenCalledTimes(1)
    })
  })

  describe('button variant', () => {
    it('is solid by default', () => {
      render(<Button label='Button' onClick={onClick} />)
      expect(screen.getByRole('button')).toHaveClass('btn-primary')
    })

    it('is SOLID', () => {
      render(<Button label='Button' onClick={onClick} variant={BUTTON_VARIANT.SOLID} />)
      expect(screen.getByRole('button')).toHaveClass('btn-primary')
    })

    it('is OUTLINE', () => {
      render(<Button label='Button' onClick={onClick} variant={BUTTON_VARIANT.OUTLINE} />)
      expect(screen.getByRole('button')).toHaveClass('btn-outline-primary')
    })

    it('is LINK', () => {
      render(<Button label='Button' onClick={onClick} variant={BUTTON_VARIANT.LINK} />)
      expect(screen.getByRole('button')).toHaveClass('btn-link')
    })

    it('is GHOST', () => {
      render(<Button label='Button' onClick={onClick} variant={BUTTON_VARIANT.GHOST} />)
      expect(screen.getByRole('button')).toHaveClass('bg-transparent')
    })
  })

  describe('button color', () => {
    it('is primary by default', () => {
      render(<Button label='Button' onClick={onClick} />)
      expect(screen.getByRole('button')).toHaveClass('btn-primary')
    })

    it('is PRIMARY', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.PRIMARY} />)
      expect(screen.getByRole('button')).toHaveClass('btn-primary')
    })

    it('is SECONDARY', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.SECONDARY} />)
      expect(screen.getByRole('button')).toHaveClass('btn-secondary')
    })

    it('is SUCCESS', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.SUCCESS} />)
      expect(screen.getByRole('button')).toHaveClass('btn-success')
    })

    it('is DANGER', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.DANGER} />)
      expect(screen.getByRole('button')).toHaveClass('btn-danger')
    })

    it('is WARNING', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.WARNING} />)
      expect(screen.getByRole('button')).toHaveClass('btn-warning')
    })

    it('is INFO', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.INFO} />)
      expect(screen.getByRole('button')).toHaveClass('btn-info')
    })

    it('is LIGHT', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.LIGHT} />)
      expect(screen.getByRole('button')).toHaveClass('btn-light')
    })

    it('is DARK', () => {
      render(<Button label='Button' onClick={onClick} color={COLORS.DARK} />)
      expect(screen.getByRole('button')).toHaveClass('btn-dark')
    })
  })

  describe('button size', () => {
    it('is small', () => {
      render(<Button label='Button' onClick={onClick} size={SIZES.SM} />)
      expect(screen.getByRole('button')).toHaveClass('btn-sm')
    })

    it('is large', () => {
      render(<Button label='Button' onClick={onClick} size={SIZES.LG} />)
      expect(screen.getByRole('button')).toHaveClass('btn-lg')
    })
  })
})
