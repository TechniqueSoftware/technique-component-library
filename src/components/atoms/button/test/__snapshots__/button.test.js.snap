// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="
    d-flex
    align-items-center
    justify-content-center
    flex-row
    button
    
    undefined
   btn btn-primary btn-md"
    role="button"
    type="button"
  >
    <span>
      Button
    </span>
  </button>
</DocumentFragment>
`;
